Description: Disable DRI on i845 and i855 Intel chips
 With DRI enabled the -intel driver on these chips is unacceptably unstable.
Forwarded: not-needed
Author: Chris Halse Rogers <christopher.halse.rogers@canonical.com>
Bug-LP: http://launchpad.net/bugs/541492
Bug-LP: http://launchpad.net/bugs/541511
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=26345
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=27187


Index: xserver-xorg-video-intel-2.9.1/src/i830_dri.c
===================================================================
--- xserver-xorg-video-intel-2.9.1.orig/src/i830_dri.c	2010-04-13 10:49:26.135418979 +1000
+++ xserver-xorg-video-intel-2.9.1/src/i830_dri.c	2010-04-13 10:56:47.775416265 +1000
@@ -385,6 +385,13 @@
     }
 #endif
 
+    if (IS_I85X (pI830) || IS_845G (pI830)) {
+	xf86DrvMsg(pScrn->scrnIndex, X_WARNING,
+		   "i845 or i855 chip detected.  Disabling DRI to prevent "
+		   "system instability.\n");
+	return FALSE;
+    }
+
     info.fd = pI830->drmSubFD;
 
     /* The whole drmOpen thing is a fiasco and we need to find a way
